abi/encode.go

```diff
@@ -161,6 +166,12 @@ func encodeTuple(v reflect.Value, t *Type) ([]byte, error) {
 		}
 	}
 
+	if t.isRootTuple && t.isDynamicType() {
+		// creates a 32 bytes offset to the head of the dynamic data
+		offset32 := append(bytes.Repeat([]byte{0}, 31), storageSlotSize)
+		ret = append(offset32, ret...)
+	}
```

Comment:
This does not seem to be correct (and probably not needed at all).

Please refer to this article, especially this part (https://medium.com/b2expand/abi-encoding-explanation-4f470927092d#:~:text=Tuples%20(dynamic%20and%20static)%20%3A).